nixos/modules/services/web-apps/matomo.nix

```diff
@@ -13,17 +14,22 @@ let
   phpExecutionUnit = "phpfpm-${pool}";
   databaseService = "mysql.service";
 
+  fqdn =
+    let
+      join = hostName: domain: hostName + optionalString (domain != null) ".${domain}";
+     in join config.networking.hostName config.networking.domain;
+
```

Comment:
This code looks correct. However when setting:
``` nix
  networking.hostName  = "jarmac"';
  networking.domain = "org";
```

I received errors from ACME that the machine was trying to register `matomo.jarmac.` which is clearly incorrect. 

Setting networking.hostname to `jarmac.org` caused an error with the prestart script looking for /var/lib/piwik, but this was resolved with another attempt, and is probably a relic from the first failed ACME attempt.

Reply:
When I designed this I did only think of `networking.hostName = "<hostname>.<domain.org>";` and `domain = null;` (what most people are probably doing today) or, imho preferrable, `networking.hostName = "<hostname>";` and `networking.domain = "<domain.org>";`. But `hostName = "jarmac";` and `domain = "org";` is nice, should really work and is what I'd like to recommend from now on for people that only want to put a single computer in their domain.

I could not reproduce the issue, though. :disappointed_relieved: The only explanation I have is that somehow the domain was set to `""`, which would explain the output. Maybe you can recheck whether that still occurs, and if yes, check what server_name landed in your nginx config (`systemctl status nginx` -> master process ngnix -c /nix/store…nginx-conf), and what landed in the acme service (should be like `systemctl status acme-matomo.jarmac..service`, and then the service file). If the issue persists, we're hopefully able to track the wrong output down that way.

Reply:
Aha! It was an error on my end:
```
  services.nginx = let
    hostname = config.networking.hostName;
    tld = config.networking.domain;
  in
  {
    enable = true;
    virtualHosts = {
      "${hostname}" = {  # here's the error, no tld splicing in
        forceSSL = true;
        enableACME = true;
        serverAliases = [ "www.${hostname}" ];
        locations."/" = {
          root = "${nginxRootPath}";
        };
      };
    };
  };
```
Apologies.

Reply:
Ok I see!

The `fqdn = let … in join config.networking.hostName config.networking.domain` snippet is actually from my `configuration.nix`, you might want a copy of that yourself so you can write `${fqdn}` everywhere. On my list of things to upstream next, as `config.lib.networking.fqdn` helper value or something. :wink: But will be quite a lot of work to find all places in nixpkgs where this should be used but isn't.